### DISCUSSION TITLE
app_state does not work when called multiple times.

### DISCUSSION THREAD

**Discussion by vdiaza** [buildnatively/community#68] [Ideas]
### Please confirm you've checked the following resources:

- [X] I've searched through the [Docs](https://docs.buildnatively.com).
- [X] I've checked [Natively Youtube Playlist 1](https://youtube.com/playlist?list=PLYo9EJqlVMNdtzb_rm6_6kjlDCmaoAqlO&si=sOygZtRDGUn_2Yi8) and [Natively Youtube Playlist 2](https://youtube.com/playlist?list=PLYo9EJqlVMNcwz2o1Xx5xVYYzzKgcyMeB&si=f4GuCPvUAdjvHplD).
- [X] I've reviewed the [Troubleshooting Guides](https://docs.buildnatively.com/guides/troubleshooting).
- [X] I've visited the [FAQ](https://docs.buildnatively.com/natively-platform/faq).

### Which web app platform are you using?

Other

### Which SDK are you using?

JavaScript SDK

### JS SDK / Bubble Plugin Version (e.g., 2.12.1)

2.12.3

### Platform with the issue?

Both

### App ID

1683332307264x280520367832039420

### Steps to reproduce

```
const appState = (callback) => {
  const info = new NativelyInfo();
  info.app_state((resp) => {
    callback(resp);
  });
};

appState(res => {
  console.log('first time')
})
appState(res => {
  console.log('second time')
})
```

after loading the app you will notice that only the "second time" is executed.

### Expected results

It would be expected to execute both functions

### Actual results

it only executes the one that was last called.

**Comment by romanfurman6** [comment]
Hello, @vdiaza,

I understand that you are inquiring about App State. Although there is no mention of this in the documentation, App State is a general instance. Therefore, it doesn't matter how many instances are created; the native app will only call the latest one that was created.

Could you please provide more details about your specific case? This will help us consider changing the current approach accordingly.
